### PR TITLE
Experiment: produce PDF with Puppeteer

### PR DESCRIPTION
**Pull request by ralfhandl**
This is currently only a sketch of how the final solution could look like.

Still missing:
- [x] parameterize with filename and variables from meta.yaml
- [x] integrate into Github workflow
- [x] only generate PDF if HTML or Markdown have changed
- [ ] extract inline styles to `odata.css`?

### CODE REVIEW

**Comment by HeikoTheissen** [comment]
> integrate into Github workflow

No need to run this with every push, unless we want to publish a PDF on github.io.

**Comment by ralfhandl** [comment]
Always having the PDF in sync with the HTML would allow spotting PDF production problems early.

File: docs/odata-data-aggregation-ext/odata-data-aggregation-ext.pdf
Comment (HeikoTheissen) [inline docs/odata-data-aggregation-ext/odata-data-aggregation-ext.pdf:1]: Bigger margin above the footer, please.

Comment (ralfhandl) [reply docs/odata-data-aggregation-ext/odata-data-aggregation-ext.pdf:1]: Please check if the additional 20% are sufficient

Comment (HeikoTheissen) [reply docs/odata-data-aggregation-ext/odata-data-aggregation-ext.pdf:1]: Have you regenerated the PDF after increasing the margin? I don't see an improvement yet.

**Review by HeikoTheissen — APPROVED** [review]

**Pull request merged**